### ISSUE TITLE
 Sparkfun Cryptoshield Testing Problem (I2C bus problem)

### ISSUE THREAD

**Issue by MiLi94** [cryptotronix/EClet#15]
I am testing my Sparkfun Cyptoshield on Arduino UNO board following the official guide which askes us to personalize the ATECC108 chip on the cryptoshield under Linux ([The beginnings of the ATECC108 ECDSA Linux driver](http://cryptotronix.com/2014/05/26/atecc108_eclet_linux_driver/)). However, when we install the EClet packet on linux and then type '`./eclet state`' on terminal, it will show the message

> "Failed to open I2C bus
> : No such file or directory". 

We then found that the directory `/dev/I2C` does not exist in our system. Could you please tell us how to solve this problem?

**Comment by jbdatko** [comment]
What development board are you using? It needs to have an i2c controller. The raspberry pi for example, you have to load the i2c kernel modules to get `/dev/i2c-X` to show up. Once you load the i2c modules then you should be able to use the driver.

**Comment by MiLi94** [comment]
I use Arduino UNO board for this project. Do we need to install i2c driver on Linux (Ubuntu) by ourself? Sorry to disturb you...

**Comment by jbdatko** [comment]
What I recommend is that you use a linux board (pi/BBB) and flywire the
i2c/pwr/gnd wires over to the shield so that you can run personalize. Then
go and use the arduino library
https://github.com/cryptotronix/cryptoauth-arduino .

The arduino library doesn't personalize the 108 so it won't do anything
useful. Which linux board are you using to flywire over i2c/pwr/gnd to the
shield?

**Comment by MiLi94** [comment]
Thank you very much. Could we use ATEML ATSAMA5D3-XPLD board to personalize the 108 chip?

**Comment by jbdatko** [comment]
Yes that what's I used actually. There's a debian image for it somewhere.
The i2c bus is different, but you can specify that on the command line to
eclet.

**Comment by MiLi94** [comment]
Sorry to disturb you again. 
Do we need to write our own i2c driver for our device? Or could we can download the i2c driver for the crytoshiled? If possible, could you please provide us with a link of i2c driver under Linux.
Many thanks

**Comment by jbdatko** [comment]
No need to write your own driver. You can use this program (eclet), you can use the user-space library I wrote (https://github.com/cryptotronix/libcrypti2c) which can then optionally be used with a kernel driver (https://github.com/cryptotronix/atsha204-i2c). You can also, use the kernel driver and then build your own application on top of it b/c it just exposes a /dev/sha0 and implements the communication layer.

**Comment by MiLi94** [comment]
Hello Josh,
We have finished the personalize work with EClet and run the ECC's example code successfully with Arduino IDE. However, we still have some problems:
1. Do we need to personalize TPM, AES use other program rather than EClet, or because we have personalized the EClet, we could directly use TPM, AES?
2. If so, how could we personalize those chips?
3. Could we use Arduion UNO to develop TPM and AES program with Arduion Library?
4. Could we use the 4 different chips in the meantime?
Thank you.

**Comment by jbdatko** [comment]
On Sat, 2016-04-16 at 07:13 -0700, MiLi94 wrote:

> Hello Josh,
> We have finished the personalize work with EClet and run the ECC's
> example code successfully with Arduino IDE. However, we still have
> some problems:
> 1. Do we need to personalize TPM, AES use other program rather than
> EClet, or because we have personalized the EClet, we could directly
> use TPM, AES?

They are all independent chips. EClet only touches the 108.

> 1. If so, how could we personalize those chips?

I have a similar linux utility (hashlet) for the 204, it's
github.com/cryptotronix/hashlet.

I don't have one for the AES chip, you have to use Atmel's code. 

> 1. Could we use Arduion UNO to develop TPM and AES program with
>    Arduion Library?

On a UNO, you probably can't use a TPM. Well, you'd have to be
creative. See: https://learn.sparkfun.com/tutorials/crypto-shield-hooku
p-guide#atmel-at97sc3204t-trusted-platform-module

You can port Atmel's code into the arduino library for the AES lib.

> 1. Could we use the 4 different chips in the meantime?

Yes? Sorry, I don't understand. The chips are all independent.
